sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -450,34 +451,24 @@ private Object getValueFromJson(String propertyName, Method method) {
    */
   @SuppressWarnings({"unchecked", "rawtypes"})
   private Object getDefault(PipelineOptions proxy, Method method) {
+    if (method.getReturnType().equals(ValueProvider.class)) {
+      // If has a default annotation, we need to supply that.
+      for (Annotation annotation : method.getAnnotations()) {
+        Object o = returnDefaultHelper(annotation, proxy, method);
+        if (o != null) {
+          return new RuntimeValueProvider(
+            method.getName(), (Class<? extends PipelineOptions>) method.getDeclaringClass(),
+            o, proxy.getOptionsId());
+        }
+      }
+      return new RuntimeValueProvider(
+        method.getName(), (Class<? extends PipelineOptions>) method.getDeclaringClass(),
+        proxy.getOptionsId());
+    }
     for (Annotation annotation : method.getAnnotations()) {
-      if (annotation instanceof Default.Class) {
-        return ((Default.Class) annotation).value();
-      } else if (annotation instanceof Default.String) {
-        return ((Default.String) annotation).value();
-      } else if (annotation instanceof Default.Boolean) {
-        return ((Default.Boolean) annotation).value();
-      } else if (annotation instanceof Default.Character) {
-        return ((Default.Character) annotation).value();
-      } else if (annotation instanceof Default.Byte) {
-        return ((Default.Byte) annotation).value();
-      } else if (annotation instanceof Default.Short) {
-        return ((Default.Short) annotation).value();
-      } else if (annotation instanceof Default.Integer) {
-        return ((Default.Integer) annotation).value();
-      } else if (annotation instanceof Default.Long) {
-        return ((Default.Long) annotation).value();
-      } else if (annotation instanceof Default.Float) {
-        return ((Default.Float) annotation).value();
-      } else if (annotation instanceof Default.Double) {
-        return ((Default.Double) annotation).value();
-      } else if (annotation instanceof Default.Enum) {
-        return Enum.valueOf((Class<Enum>) method.getReturnType(),
-            ((Default.Enum) annotation).value());
-      } else if (annotation instanceof Default.InstanceFactory) {
-        return InstanceBuilder.ofType(((Default.InstanceFactory) annotation).value())
-            .build()
-            .create(proxy);
+      Object o = returnDefaultHelper(annotation, proxy, method);
```

Comment:
Can we clean up this logic? Something like this seems easier to reason about, but may not be exactly right.

```java
@Nullable Object default;
for (Annotation annotation ; method.getAnnotations()) {
    default = getDefaultHelper(annotation, proxy, method);
    if (default != null) {
        break;
    }
}
if (method.getReturnType().equals(VP.class)) { // or whatever we change to above
    return default == null
        ? new RVP(...)
        : new RVP(...);
}
else if (default != null) {
    return default;
} else {
    return Defaults.defaultValue(method.getReturnType());
}
```